custom_components/solplanet/coordinator.py

```diff
@@ -51,7 +51,7 @@ async def _async_update_data(self):
             )
 
             meter = None
-            meter_sn = isns[0]
+            meter_sn = isns[0] if len(isns) > 1 else None
```

Comment:
Condition should check if we have any item in isns array.

```suggestion
            meter_sn = isns[0] if len(isns) > 0 else None
```

Reply:
In such setups where there is no smart meter installed (like mine), there is only 1 serial in isns while inverter is operational, and there is 0 when inverter is in sleep mode. If conditon would check for > 0 then a smart meter component is created with the serial of the inverter. Although > 1 is perfect in my setup, maybe it would have unwanted behaviour when one inverter and meter are installed without battery and in sleep mode only one 1 serial would appear.

Reply:
Right now it's correct to create meter with isn same as inverter. When we use CT clamps as a meter, there is no `sn` field in meter info endpoint (http://ip:8484/getdev.cgi?device=3). The situation is different for the Eastron meters - in that case it returns its serial number which can be used to index, but it's not a requirement as this index is used to create different sensor name in HA.